src/poetry/core/packages/file_dependency.py

```diff
@@ -43,10 +35,40 @@ def __init__(
             optional=optional,
             allows_prereleases=True,
             source_type="file",
-            source_url=self._full_path.as_posix(),
+            source_url=full_path.as_posix(),
             extras=extras,
         )
 
+    def _get_full_path(self, raise_if_invalid: bool, name: str) -> Path:
```

Comment:
Adding `name` was necessary because `.name` doesn't get set until `super().__init__` is called which already requires `full_path`.